File: server/src/main/java/com/example/server/dto/Album/AlbumDto.java
package com.example.server.dto.Album;

import com.example.server.dto.Author.AuthorSimpleDto;
import com.example.server.dto.Track.TrackSimpleDto;
import lombok.Data;

import java.util.Set;

@Data
public class AlbumDto
{
    private Long id;
    private String title;
    private String imgUrl;
    private Set<AuthorSimpleDto> authors;
    private String releaseDate;
    private Set<TrackSimpleDto> tracks;
}

File: client/src/models/AlbumDto.ts
import type {AuthorSimpleDto} from "./DTO/AuthorSimpleDto.ts";
import type {TrackSimpleDto} from "./DTO/track/TrackSimpleDto.ts";

export interface AlbumDto {
    id: number;
    title: string;
    imgUrl: string;
    authors: AuthorSimpleDto[];
    releaseDate: Date;
    tracks: TrackSimpleDto[];
}